File: app/src/prod/java/com/davismiyashiro/expenses/Injection.java
package com.davismiyashiro.expenses;

import android.content.Context;

import com.davismiyashiro.expenses.model.InMemoryTabRepository;
import com.davismiyashiro.expenses.model.TabRepository;
import com.davismiyashiro.expenses.model.localrepo.TabRepositoryDataSourceLocal;

/**
 * Created by Davis on 25/01/2016.
 */
public class Injection {

    public static TabRepository provideTabsRepository(Context context) {
        return InMemoryTabRepository.getInstance(TabRepositoryDataSourceLocal.getInstance(context));
    }

}

File: app/src/prod/java/com/davismiyashiro/expenses/Injection.kt
package com.davismiyashiro.expenses

import android.content.Context

import com.davismiyashiro.expenses.model.RepositoryImpl
import com.davismiyashiro.expenses.model.Repository
import com.davismiyashiro.expenses.model.localrepo.RepositoryDataSourceLocal

/**
 * Created by Davis on 25/01/2016.
 */
object Injection {

    fun provideTabsRepository(context: Context): Repository {
        return RepositoryImpl.getInstance(RepositoryDataSourceLocal.getInstance(context))
    }

}
